### PR TITLE
Whiteboard Jira 1298 Changing avatars causes paint sphere to be on wrong finger

### PR DESCRIPTION
**Pull request by RebeccaStankus**
https://highfidelity.atlassian.net/browse/BUGZ-1298
https://highfidelity.atlassian.net/browse/BUGZ-1206

### CODE REVIEW

File: usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js [hunk 1]
```diff
@@ -137,6 +129,29 @@
             _this.playSound(OPEN_SOUND, OPEN_SOUND_VOLUME, MyAvatar.position, true, false);
         },
 
+        /* Collect IDs of integral parts used in drawing */
+        prepareDrawingData: function() {
+            try {
+                var properties = Entities.getEntityProperties(_this.entityID, ['userData','color']);
+                var parsedUserData = JSON.parse(properties.userData);
+            } catch (err) {
+                print("ERROR: Could not parse userData of color sphere.");
```
Comment (zfox23) [inline usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js]: you should return after this, right? will the thing work if we continue here?

File: usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js [hunk 2]
```diff
@@ -252,8 +249,9 @@
                 lineProperties.color = _this.color;
                 lineProperties.textures = _this.texture;
                 lineProperties.faceCamera = !onBoard;
-                if (polyLine && whiteboard) {
-                    Entities.editEntity(polyLine, { parentID: whiteboard });
+                if (polyLine && _this.whiteboard) {
+                    Entities.editEntity(polyLine, { parentID: _this.whiteboard });
+
```
Comment (zfox23) [inline usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js]: remove extra whitespace please

File: usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js [hunk 3]
```diff
@@ -514,9 +512,10 @@
             Script.setTimeout(function(){
                 _this.playSound(DRAW_SOUND, DRAW_SOUND_VOLUME, lineStartPosition, true, true);
             }, SOUND_DELAY_TIME);
-
-            isTheTriggerUpdateRunning = true;
-            Script.update.connect(_this.onTriggerPressedScriptUpdate);
+            if (!_this.isTheTriggerUpdateRunning) {
+                Script.update.connect(_this.onTriggerPressedScriptUpdate);
```
Comment (zfox23) [inline usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js]: I didn't realize we were using the update loop in this script. Why does this script need to use the update loop? Why can't it use a controller mapping?

File: usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js [hunk 4]
```diff
@@ -532,42 +531,44 @@
...
+            if (!_this.isUserInZone(_this.whiteboardZone)) {
                 Entities.deleteEntity(_this.entityID);
             }
             if (tablet.tabletShown || activeTriggerPress) {
                 return;
             }
-            isTheGripUpdateRunning = true;
-            Script.update.connect(_this.onGripPressedScriptUpdate);
+            if (!_this.isTheGripUpdateRunning) {
+                Script.update.connect(_this.onGripPressedScriptUpdate);
```
Comment (zfox23) [inline usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js]: same deal - why can't we use a controller mapping? two handlers connected to the update loop in one script is a recipe for serious slowdown

File: usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js [hunk 5]
```diff
@@ -732,6 +733,7 @@
 
         /* Delete paint sphere in hand when switching dominant hand */
         handChanged: function() {
+            Entities.callEntityMethod(_this.colorPaletteID, "createPaintSphere");
```
Comment (zfox23) [inline usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js]: does it matter that this call is before the `deleteEntity()` call below?

File: usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js [hunk 6]
```diff
@@ -829,10 +833,14 @@
 
         onScaleChanged: function(){
             var sphereProperties = Entities.getEntityProperties(_this.entityID, ['dimensions']);
-        
             currentStrokeWidth = sphereProperties.dimensions.x;
         },
 
+        onSkeletonModelURLChanged: function(){
+            Entities.callEntityMethod(_this.colorPaletteID, "createPaintSphere");
```
Comment (zfox23) [inline usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js]: does it matter that this call is before the deleteEntity() call below?

File: usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js (hunk 1 above)
Comment (RebeccaStankus) [reply usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js]: oh, I meant to add a respawn of the paint sphere there. Will add that in.

File: usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js (hunk 5 above)
Comment (RebeccaStankus) [reply usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js]: It was supposed to be so to ensure it happens before this paint sphere gets deleted.

File: usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js (hunk 6 above)
Comment (RebeccaStankus) [reply usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js]: It was supposed to be so to ensure it happens before this paint sphere gets deleted.

Comment (zfox23) [reply usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js]: OK - I _think_ that is safe.

File: usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js (hunk 5 above)
Comment (zfox23) [reply usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js]: OK - I _think_ that is safe.

File: usefulUtilities/airAndWhiteboardDraw/README.md [hunk 7]
```diff
@@ -12,37 +12,37 @@ This utility combines the best features of the whiteboard and the draw app. Any
 
 ## Releases
 
-SHA [](https://github.com/highfidelity/hifi-content/commits/)
+SHA [71386f1](https://github.com/highfidelity/hifi-content/pull/455/commits/71386f1)
```
Comment (zfox23) [inline usefulUtilities/airAndWhiteboardDraw/README.md]: Please change all of these links to match the format https://github.com/highfidelity/hifi-content/commits/<commit SHA>

**Pull request merged**